### ISSUE TITLE
UIA in Windows consoles: Say all behaves incorrectly

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#9875]
### Steps to reproduce:
1. Turn on UIA in console
1. Open console
1. Press NVDA+up arrow twice fast
1. Press NVDA+down arrow

### Actual behavior:
NVDA walks backwards through the text, speaking one extra letter at a time.
```
>
f>
ef>
eef>
Reef>
\Reef>
s\Reef>
rs\Reef>
ers\Reef>
>
f>
ef>
eef>
Reef>
\Reef>
s\Reef>
rs\Reef>
ers\Reef>
C:\Users\Reef>
```

### Expected behavior:
Standard "say all" behaviour.

#### NVDA version:
Running from source on `threshold_py3_staging` 
`beta-17939,fd65a653` seems to freeze.

#### Windows version:
Windows 10 - 1903
Build: 18362.175

#### Other information:

Quite possibly related to the textUtils offsets work.

**Comment by leonardder** [comment]
The uniscribe code in there needs to be updated to use textUtils.

**Comment by leonardder** [comment]
@feerrenrut: Just to make sure, don't you have this issue on master? I don't see how it could be related to uniscribe actually, as say all moves by line and uniscribe is only used for per word movement.

**Comment by leonardder** [comment]
Just confirmed that this also affects master and threshold.

**Comment by codeofdusk** [comment]
This is caused by a bug in the `ConsoleUIATextInfo.collapse` method. A PR is pending which will restore say all functionality to the level present in legacy consoles (i.e. say all in review is implemented but NVDA+down arrow isn't).

**Comment by leonardder** [comment]
Fixed in #9887
